pennylane/resources/first_quantization.py

```diff
@@ -15,505 +15,558 @@
 
-def _cost_qrom(lz):
-    r"""Return the minimum number of Toffoli gates needed for erasing the output of a QROM.
+                \lambda_{\nu} = \sum_{\nu \in G_0} \frac{1}{\left \| \nu \right \|^2},
 
-    Args:
-        lz (int): sum of the atomic numbers
+            we follow Eq. (F6) of
+            `PRX 8, 011044 (2018) <https://journals.aps.org/prx/abstract/10.1103/PhysRevX.8.011044>`_
+            and use
 
-    Returns:
-        int: the minimum cost of erasing the output of a QROM
+            .. math::
 
-    **Example**
+                \lambda_{\nu} = 4\pi \left ( \frac{\sqrt{3}}{2} N^{1/3} - 1 \right) + 3 - \frac{3}{N^{1/3}}
+                + 3 \int_{x=1}^{N^{1/3}} \int_{y=1}^{N^{1/3}} \frac{1}{x^2 + y^2} dydx.
 
-    >>> _cost_qrom_min(100)
-    21
-    """
-    if lz <= 0 or not isinstance(lz, int):
-        raise ValueError("The sum of the atomic numbers must be a positive integer.")
+            We also need to compute :math:`\lambda^{\alpha}_{\nu}` defined in Eq. (123) of
+            `PRX Quantum 2, 040332 (2021) <https://link.aps.org/doi/10.1103/PRXQuantum.2.040332>`_
 
-    k_f = np.floor(np.log2(lz) / 2)
-    k_c = np.ceil(np.log2(lz) / 2)
+            .. math::
 
-    cost_f = int(2**k_f + np.ceil(2 ** (-1 * k_f) * lz))
-    cost_c = int(2**k_c + np.ceil(2 ** (-1 * k_c) * lz))
+                \lambda^{\alpha}_{\nu} = \alpha \sum_{\nu \in G_0} \frac{\left \lceil
+                \mathcal{M}(2^{\mu - 2}) / \left \| \nu \right \|^2 \right \rceil}{\mathcal{M}
+                2^{2\mu - 4}},
 
-    return min(cost_f, cost_c)
+            which we compute here, for :math:`\alpha = 1`, as
 
+            .. math::
 
-def unitary_cost(n, eta, omega, error, lamb, br=7, charge=0):
-    r"""Return the number of Toffoli gates needed to implement the qubitization unitary operator.
+                \lambda^{1}_{\nu} = \lambda_{\nu} + \epsilon_l,
 
-    The expression for computing the cost is taken from Eq. (125) of
-    [`PRX Quantum 2, 040332 (2021) <https://link.aps.org/doi/10.1103/PRXQuantum.2.040332>`_].
+            where :math:`\epsilon_l` is simply defined as the difference of
+            :math:`\lambda^{1}_{\nu}` and :math:`\lambda_{\nu}`. We follow Eq. (113) of
+            `PRX Quantum 2, 040332 (2021) <https://link.aps.org/doi/10.1103/PRXQuantum.2.040332>`_
+            to derive an upper bound for its absolute value:
 
-    Args:
-        n (int): number of plane waves
-        eta (int): number of electrons
-        omega (float): unit cell volume
-        error (float): target error in the algorithm
-        lamb (float): 1-norm of the Hamiltonian
-        br (int): number of bits for ancilla qubit rotation
-        charge (int): total electric charge of the system
+            .. math::
 
-    Returns:
-        int: the number of Toffoli gates needed to implement the qubitization unitary operator
+                \epsilon_l = \frac{4}{2^{n_m}} (7 \times 2^{n_p + 1} + 9 n_p - 11 - 3 \times 2^{-n_p}),
```

Comment:
perhaps it should be |\epsilon_l| \le  since it's an upper bound for the absolute value of the difference.